### PR TITLE
Extract specific transaction pool errors.

### LINKED ISSUE #1806: Better error response from RPC for failed transaction
Currently if a transaction failed, all I can get is from polkadot.js is `error.message [-32603]: Unknown error occured`
It should be possible to give a more detailed error message.

This RPC response, which does indicate the error code 3, which is `CantPay`, but it is hard to get it out of the data string.

```
{
    "code": -32603,
    "message": "Unknown error occured",
    "data": "Error(Pool(InvalidTransaction(3)), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })"
}
```

https://github.com/paritytech/substrate/blob/6a4b58068421f280ff2dfe564220c58147c35f4a/core/sr-primitives/src/lib.rs#L331

However the data field needs to be more parseable and the correct error code should be returned (maybe in a subcode field?)

cc @jacogr

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #1806

### CODE REVIEW

File: core/rpc/src/author/error.rs [hunk 1]
```diff
@@ -63,6 +63,41 @@ impl From<Error> for rpc::Error {
 				message: e.description().into(),
 				data: Some(format!("{:?}", e).into()),
 			},
+			Error(ErrorKind::Pool(txpool::error::ErrorKind::InvalidTransaction(code)), _) => rpc::Error {
+				code: rpc::ErrorCode::ServerError(ERROR + 3),
```
Comment (bkchr) [inline core/rpc/src/author/error.rs]: Are these codes used for something?

Comment (tomusdrw) [reply core/rpc/src/author/error.rs]: They can be arbitrary and it's up to us to define their meaning. They should just be unique, I suppose they are going to be useful for the consumers of the API.

Comment (bkchr) [reply core/rpc/src/author/error.rs]: If they are useful for the consumers of the API, wouldn't it be better to put them into constants?

Comment (tomusdrw) [reply core/rpc/src/author/error.rs]: Updated

Comment (bkchr) [reply core/rpc/src/author/error.rs]: Ty

**Pull request merged**